suite-native/module-add-accounts/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,194 @@
+    Object.keys(networks).forEach(symbol =>
+        availableNetworkAccountTypes.set(symbol as NetworkSymbol, [
+            NORMAL_ACCOUNT_TYPE,
+            ...(Object.keys(networks[symbol].accountTypes) as AccountType[]).filter(
+                accountType => accountType !== 'coinjoin', // coinjoin is not supported
+            ),
+        ]),
+    );
+
+    const getAvailableAccountTypesForNetwork = ({ network }: { network: Network }) =>
+        availableNetworkAccountTypes.get(network.symbol) ?? [NORMAL_ACCOUNT_TYPE];
+
+    const getDefaultAccountType = ({ network }: { network: Network }) =>
+        getAvailableAccountTypesForNetwork({ network })[0];
+
+    const getNetwork = ({
+        networkSymbol,
+        accountType,
+    }: {
+        networkSymbol: NetworkSymbol;
+        accountType?: AccountType;
+    }) => {
+        const type =
+            accountType ??
+            getDefaultAccountType({
+                network: supportedNetworks.filter(network => network.symbol === networkSymbol)[0],
+            });
+
+        return supportedNetworks.filter(
+            network =>
+                network.symbol === networkSymbol &&
+                (type === NORMAL_ACCOUNT_TYPE
+                    ? network.accountType === undefined
+                    : network.accountType === type),
+        )[0];
+    };
+
+    const showTooManyAccountsAlert = () =>
+        showAlert({
+            title: translate('moduleAddAccounts.alerts.tooManyAccounts.title'),
+            description: translate('moduleAddAccounts.alerts.tooManyAccounts.description'),
+            icon: 'warningCircle',
+            pictogramVariant: 'red',
+            primaryButtonTitle: translate('moduleAddAccounts.alerts.tooManyAccounts.actionPrimary'),
+            onPressPrimaryButton: () => {
+                hideAlert();
+            },
+        });
+
+    const showAnotherEmptyAccountAlert = () =>
+        showAlert({
+            title: translate('moduleAddAccounts.alerts.anotherEmptyAccount.title'),
+            description: translate('moduleAddAccounts.alerts.anotherEmptyAccount.description'),
+            icon: 'warningCircle',
+            pictogramVariant: 'red',
+            primaryButtonTitle: translate(
+                'moduleAddAccounts.alerts.anotherEmptyAccount.actionPrimary',
+            ),
+            onPressPrimaryButton: () => {
+                hideAlert();
+            },
+            secondaryButtonTitle: translate(
+                'moduleAddAccounts.alerts.anotherEmptyAccount.actionSecondary',
+            ),
+            onPressSecondaryButton: () => {
+                openLink(
+                    translate('moduleAddAccounts.alerts.anotherEmptyAccount.actionSecondaryUrl'),
+                );
+                hideAlert();
+            },
+        });
+
+    const addCoinAddress = ({ network, type }: { network: Network; type?: AccountType }) => {
```

Comment:
Name seems strange, it's not adding address but account right? Account can have multiple addresses.

Reply:
454699331c